### PR TITLE
[site] Enhance the external link checking

### PR DESCRIPTION
**Pull request by dshevtsov**
## Purpose of this pull request

This pull request enhances the link checking functionality on devdocs:
- Enables caching for external links
- Associates devdocs.magento.com with internal domains for more strict check. This allows to check links that are redirected on AWS
- Excludes files in `guides/v2.3/mrg/*` from link checking, because the files are generated from the codebase and all the fixes have to be done into the Magento codebase
- Fixes/removes the links to make the build pass tests
- Customizes the template for redirects (the jekyll-redirect-from plugin) to exclude the generated HTML files with redirects from link checking
- **The external links checking remains disabled** (fro more details, read conversation in this pull request)

## Additional information

This also changes the source branch for the MFTF documentation from `develop` to `master`

### CODE REVIEW

File: guides/v2.3/extension-dev-guide/depend-inj.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 ---
 group: php-developer-guide
 title: Dependency injection
-menu_title: Dependency injection
+redirect_from:
+- guides/v2.0/extension-dev-guide/depend-inj.html
```
Comment (keharper) [inline guides/v2.3/extension-dev-guide/depend-inj.md]: Remove this.

File: guides/v2.2/architecture/archi_perspectives/present_layer.md [hunk 2]
```diff
@@ -29,14 +28,6 @@ Each [theme](https://glossary.magento.com/theme) resides in a unique directory a
...
 
-## GraphQL
-
-GraphQL is a data query language developed internally by Facebook in 2012 before being publicly released in 2015. Magento implements GraphQL to provide an alternative to REST and SOAP web APIs for frontend development.
-
-GraphQL allows you to define the structure of the data that you need, and the server returns only the data you request. Each GraphQL-capable module contains a declarative schema that defines the syntax for queries that the module supports, as well as the attributes that can be returned. If you run a REST call such as GET /V1/products/:sku on a simple product, the system might fetch more than 100 lines of data. If all you need is the current price, the call has returned significantly more information than you need. With GraphQL, a query against the same SKU could return just the price.
-
-More information can be found in the [GraphQL Developer Guide][]
-
 ## Progressive Web Apps
```
Comment (keharper) [inline guides/v2.2/architecture/archi_perspectives/present_layer.md]: @jcalcaben Should PWA be mentioned in a 2.2 document?

**Comment by dshevtsov** [comment]
running tests

File: guides/v2.2/architecture/archi_perspectives/present_layer.md (hunk 2 above)
Comment (jcalcaben) [reply guides/v2.2/architecture/archi_perspectives/present_layer.md]: no, PWA Studio is 2.3+

Comment (dshevtsov) [reply guides/v2.2/architecture/archi_perspectives/present_layer.md]: This is out of scope in this pull request. But I removed the sections about PWA, there were two of them with the same heading. Also, I removed the duplicated section in 2.3.

**Comment by dshevtsov** [comment]
I disabled the external link checking, because some servers responded with 429 code (Too Many Requests). Considering this, we should not test external links that often as we test internal links. The test should be run periodically, once a week or once a month. Maybe will find a way to run external link checking for newly introduced links only for each PR.

@jeff-matthews the PR can be safely unblocked so that I could merge all the fixed links that it contains.

**Comment by dshevtsov** [comment]
We also have other options: we can ignore 429 status codes, or disable link checking for the domains that return 429.

**Comment by hguthrie** [comment]
Checking the newly introduced links sounds very reasonable.

**Comment by dshevtsov** [comment]
running tests

**Pull request merged**